addons/mail/static/src/core/common/message.js

```diff
@@ -436,18 +436,30 @@ export class Message extends Component {
      */
     prepareMessageBody(element) {}
 
-    enterEditMode() {
+    async enterEditMode() {
+        const body = document.createElement("span");
+        body.innerHTML = this.props.message.body;
         const messageContent = convertBrToLineBreak(this.props.message.body);
-        this.props.message.composer = {
-            mentionedPartners: this.props.message.recipients,
-            textInputContent: messageContent,
-            selection: {
-                start: messageContent.length,
-                end: messageContent.length,
-                direction: "none",
-            },
-        };
-        this.state.isEditing = true;
+        const mentionedChannelElements = body.querySelectorAll(".o_channel_redirect");
+        const mentionedChannels = Array.from(mentionedChannelElements).map(async (el) =>
+            this.store.Thread.getOrFetch({
+                id: el.dataset.oeId,
+                model: el.dataset.oeModel,
+            })
+        );
+        Promise.all(mentionedChannels).then((mentionedChannels) => {
+            this.props.message.composer = {
+                mentionedChannels,
+                mentionedPartners: this.props.message.recipients,
+                textInputContent: messageContent,
+                selection: {
+                    start: messageContent.length,
+                    end: messageContent.length,
+                    direction: "none",
+                },
+            };
+            this.state.isEditing = true;
+        });
```

Comment:
Editing can take seconds to effectively start.

I would solve this small problem by showing spinner at place of message actions when not mouse-hovering the message.

Where message actions are displayed, only on non-mouse hover so that message actions can still be accessed all the time like before, but at the very least there's some visual to tell it takes time where the eyes were when clicking from message edit.

Note that a start editing could be canceled by user clicking on edit or reply on another message. Code should be robust against them.

Reply:
Ideally editing should not be delayed whatsoever by this feature.

We only need the list of "initial" channels when actually "saving" the edit, not when displaying it.

The only thing that should be delayed is the save if it actually happens.

Reply:
Hello, I think the ideal way of doing this should be for the `getOrFetch` to happen the same time while the user is editing and it is not blocking the workflow. So I would suggest to do as what I pushed and you can have a look at that.

Reply:
> Hello, I think the ideal way of doing this should be for the `getOrFetch` to happen the same time while the user is editing and it is not blocking the workflow.

Good idea indeed, and if the user edits very quickly before RPC then it's a bit slow to save but I think this is better tradeoff :)